### PR TITLE
Optimize CDQOp

### PR DESCRIPTION
**Pull request by pmatos**
Optimize `cwde` and `cbw`.

Change OpcodeDispatcher CDQOp to load directly from GPR, and recognize instruction width.

Also change Constant Propagation to synthesize 32bits constant when necessary.

Update instcountci files.

### CODE REVIEW

File: FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp [hunk 1]
```diff
@@ -1395,11 +1395,12 @@ void OpDispatchBuilder::XCHGOp(OpcodeArgs) {
 }
 
 void OpDispatchBuilder::CDQOp(OpcodeArgs) {
-  OrderedNode *Src = LoadSource(GPRClass, Op, Op->Src[0], Op->Flags);
   uint8_t DstSize = GetDstSize(Op);
   uint8_t SrcSize = DstSize >> 1;
+  OrderedNode *Src = LoadGPRRegister(X86State::REG_RAX, SrcSize, 0, true);
```
Comment (alyssarosenzweig) [inline FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp:1400]: Why did you change this? We typically use LoadSource

Comment (alyssarosenzweig) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp:1400]: If you  want to eliminate the masking, what you want is `{.AllowUpperGarbage = true}` which opts into loading garbage values in the upper bits.

Comment (pmatos) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp:1400]: > If you want to eliminate the masking, what you want is `{.AllowUpperGarbage = true}` which opts into loading garbage values in the upper bits. See #3381 as an example

But `LoadSource` takes `const LoadSourceOptions& Options`, while `LoadGPRRegister` takes a `bool AllowUpperGarbage`.

**Comment by pmatos** [comment]
instcountci is failing - apparently it's using the old json file, not the one I just committed. Is there any procedure to properly submit a json file to change instcountci?

**Comment by Sonicadvance1** [comment]
> instcountci is failing - apparently it's using the old json file, not the one I just committed. Is there any procedure to properly submit a json file to change instcountci?

Looks like in this case you didn't get the full json modified. Specifically the `ExpectedInstructionCount` fields of each should have been changed.
Doing a `ninja instcountci_tests` and then `ninja instcountci_update_tests` should change that?

File: FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp (hunk 1 above)
Comment (Sonicadvance1) [reply FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp:1400]: Either route is okay to take. In a lot of cases for hardcoded register usage inside of an instruction we will use `LoadGPRRegister`. Since the cwd family of instructions hardcode their register usage to RAX then I think it is fine to use `LoadGPRRegister`.

If the instruction wasn't hardcoded to a register then `LoadSource` is definitely preferred if possible.

**Comment by pmatos** [comment]
> > instcountci is failing - apparently it's using the old json file, not the one I just committed. Is there any procedure to properly submit a json file to change instcountci?
> 
> Looks like in this case you didn't get the full json modified. Specifically the `ExpectedInstructionCount` fields of each should have been changed. Doing a `ninja instcountci_tests` and then `ninja instcountci_update_tests` should change that?

Odd - you're right. instruction counts were not modified. Pretty sure I already did update_tests... I mean, that's how Primary.json got modified. Let me try that again.

**Comment by pmatos** [comment]
> > > instcountci is failing - apparently it's using the old json file, not the one I just committed. Is there any procedure to properly submit a json file to change instcountci?
> > 
> > 
> > Looks like in this case you didn't get the full json modified. Specifically the `ExpectedInstructionCount` fields of each should have been changed. Doing a `ninja instcountci_tests` and then `ninja instcountci_update_tests` should change that?
> 
> Odd - you're right. instruction counts were not modified. Pretty sure I already did update_tests... I mean, that's how Primary.json got modified. Let me try that again.

Now it worked - maybe I ran things improperly first time around.

**Review by lioncash — APPROVED** [review]
